Assemblers.Common/PackageReferenceProcessor.cs

```diff
@@ -184,13 +179,14 @@ private static bool IsDevPackNuGetPackage(string packageId)
         /// <param name="extensions">The NuGet packages.</param>
         /// <param name="allPackagesAndDependencies">All top level packages and their dependencies.</param>
         /// <returns>The resolved NuGet packages.</returns>
-        private static IEnumerable<SourcePackageDependencyInfo> GetResolvedPackages(ISourceRepositoryProvider sourceRepositoryProvider,
+        private static IEnumerable<PackageIdentity> GetResolvedPackages(ISourceRepositoryProvider sourceRepositoryProvider,
                                                                               ILogger logger, IList<PackageIdentity> extensions,
                                                                               HashSet<SourcePackageDependencyInfo> allPackagesAndDependencies)
         {
             if (allPackagesAndDependencies.Count == 0)
             {
-                return Array.Empty<SourcePackageDependencyInfo>();
+                // If no dependencies, then return the main package(s) at least.
+                return extensions;
```

Comment:
Not sure I understand this. Can you explain the logic?

Reply:
This method returns the list of all the NuGet packages that are needed. If there are no dependencies, it would return an empty list, resulting in nothing. As for SRM we ignore all the dependencies, if you only have that NuGet, then it would return nothing. So the further logic wouldn't know about it and would be unable to add the script reference or dllImport.